jni/src/knn_extension/faiss/utils/BitSet.cpp

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <assert.h>
+#include "BitSet.h"
+
+FixedBitSet::FixedBitSet(const int* int_array, const int length){
+    assert(int_array && "int_array should not be null");
+    int maxValue = int_array[length - 1];
```

Comment:
To find the max value in the int_array shouldn't you iterate over the array and then find the max value. Here we are taking the assumption that the last element will always have the max value.

Reply:
+1, also we need to check for `length`, if it's not 0.

Reply:
Let me iterate the array.